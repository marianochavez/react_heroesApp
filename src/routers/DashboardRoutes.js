import React from 'react';
import { Routes, Route } from "react-router-dom";
import { MarvelScreen } from '../components/marvel/MarvelScreen';
import { DcScreen } from '../components/dc/DcScreen';
import { SearchScreen } from '../components/search/SearchScreen';
import { Navbar } from '../components/ui/NavBar';
import { HeroScreen } from '../components/hero/HeroScreen';

export const DashboardRoutes = () => {
  return (
        <>
            <Navbar/>
            <div className='container'>
                <Routes>
                    <Route path="marvel" element={<MarvelScreen />} />
                    <Route path="dc" element={<DcScreen />} />

                    <Route path="search" element={<SearchScreen />} />
                    <Route path="hero/:heroId" element={<HeroScreen />} />

                    <Route path="/" element={<MarvelScreen />} />
                </Routes>
            </div>
        </>
    );
};
