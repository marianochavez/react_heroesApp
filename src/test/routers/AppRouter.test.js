import {mount} from 'enzyme';
import { AuthContext } from '../../auth/authContext';
import { AppRouter } from "../../routers/AppRouter";

describe('AppRouter tests', () => {
  
    const contextValue = {
      user: {
          logged: false
      }
      };

    test('should display login if user is unauthenticated', () => {

        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter />
            </AuthContext.Provider>
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('h1').text().trim()).toBe('Login');
    });
    
    test('should display marvel component if is autrhenticated', () => {

        const contextValue = {
            user: {
                logged: true,
                name: 'Pepe'
            }
        };

        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter />
            </AuthContext.Provider>
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.navbar').exists()).toBeTruthy();
    });
    
});
