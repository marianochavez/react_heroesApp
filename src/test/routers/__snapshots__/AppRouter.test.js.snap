// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRouter tests should display login if user is unauthenticated 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;

exports[`AppRouter tests should display marvel component if is autrhenticated 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Heroes
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          Pepe
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div>
      <h1>
        MarvelScreen
      </h1>
      <hr />
      <div
        className="row rows-cols-1 rows-cols-md-3 g-3 
 animate__animated animate__fadeIn"
      >
        <div
          className="col-4 animate__animated animate__fadeIn"
        >
          <div
            className="card mb-3"
          >
            <div
              className="row g-0"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Spider Man"
                  className="card-img"
                  src="/assets/marvel-spider.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Spider Man
                  </h5>
                  <p
                    className="card-text"
                  >
                    Peter Parker
                  </p>
                  <p
                    className="cart-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Amazing Fantasy #15
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-spider"
                    onClick={[Function]}
                  >
                    See more
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-4 animate__animated animate__fadeIn"
        >
          <div
            className="card mb-3"
          >
            <div
              className="row g-0"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Captain America"
                  className="card-img"
                  src="/assets/marvel-captain.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Captain America
                  </h5>
                  <p
                    className="card-text"
                  >
                    Steve Rogers
                  </p>
                  <p
                    className="cart-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Captain America Comics #1
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-captain"
                    onClick={[Function]}
                  >
                    See more
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-4 animate__animated animate__fadeIn"
        >
          <div
            className="card mb-3"
          >
            <div
              className="row g-0"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Iron Man"
                  className="card-img"
                  src="/assets/marvel-iron.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Iron Man
                  </h5>
                  <p
                    className="card-text"
                  >
                    Tony Stark
                  </p>
                  <p
                    className="cart-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Tales of Suspense #39
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-iron"
                    onClick={[Function]}
                  >
                    See more
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-4 animate__animated animate__fadeIn"
        >
          <div
            className="card mb-3"
          >
            <div
              className="row g-0"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Thor"
                  className="card-img"
                  src="/assets/marvel-thor.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Thor
                  </h5>
                  <p
                    className="card-text"
                  >
                    Thor Odinson
                  </p>
                  <p
                    className="cart-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Journey into Myster #83
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-thor"
                    onClick={[Function]}
                  >
                    See more
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-4 animate__animated animate__fadeIn"
        >
          <div
            className="card mb-3"
          >
            <div
              className="row g-0"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Hulk"
                  className="card-img"
                  src="/assets/marvel-hulk.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Hulk
                  </h5>
                  <p
                    className="card-text"
                  >
                    Bruce Banner
                  </p>
                  <p
                    className="cart-text"
                  >
                    <small
                      className="text-muted"
                    >
                      The Incredible Hulk #1
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-hulk"
                    onClick={[Function]}
                  >
                    See more
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-4 animate__animated animate__fadeIn"
        >
          <div
            className="card mb-3"
          >
            <div
              className="row g-0"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Wolverine"
                  className="card-img"
                  src="/assets/marvel-wolverine.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Wolverine
                  </h5>
                  <p
                    className="card-text"
                  >
                    James Howlett
                  </p>
                  <p
                    className="cart-text"
                  >
                    <small
                      className="text-muted"
                    >
                      The Incredible Hulk #180
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-wolverine"
                    onClick={[Function]}
                  >
                    See more
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-4 animate__animated animate__fadeIn"
        >
          <div
            className="card mb-3"
          >
            <div
              className="row g-0"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Daredevil"
                  className="card-img"
                  src="/assets/marvel-daredevil.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Daredevil
                  </h5>
                  <p
                    className="card-text"
                  >
                    Matthew Michael Murdock
                  </p>
                  <p
                    className="cart-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Daredevil #1
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-daredevil"
                    onClick={[Function]}
                  >
                    See more
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-4 animate__animated animate__fadeIn"
        >
          <div
            className="card mb-3"
          >
            <div
              className="row g-0"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Hawkeye"
                  className="card-img"
                  src="/assets/marvel-hawkeye.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Hawkeye
                  </h5>
                  <p
                    className="card-text"
                  >
                    Clinton Francis Barton
                  </p>
                  <p
                    className="cart-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Tales of Suspense #57
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-hawkeye"
                    onClick={[Function]}
                  >
                    See more
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-4 animate__animated animate__fadeIn"
        >
          <div
            className="card mb-3"
          >
            <div
              className="row g-0"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Cyclops"
                  className="card-img"
                  src="/assets/marvel-cyclops.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Cyclops
                  </h5>
                  <p
                    className="card-text"
                  >
                    Scott Summers
                  </p>
                  <p
                    className="cart-text"
                  >
                    <small
                      className="text-muted"
                    >
                      X-Men #1
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-cyclops"
                    onClick={[Function]}
                  >
                    See more
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-4 animate__animated animate__fadeIn"
        >
          <div
            className="card mb-3"
          >
            <div
              className="row g-0"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Silver Surfer"
                  className="card-img"
                  src="/assets/marvel-silver.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Silver Surfer
                  </h5>
                  <p
                    className="card-text"
                  >
                    Norrin Radd
                  </p>
                  <p
                    className="cart-text"
                  >
                    <small
                      className="text-muted"
                    >
                      The Fantastic Four #48
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-silver"
                    onClick={[Function]}
                  >
                    See more
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
